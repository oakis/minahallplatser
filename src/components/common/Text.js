import React from 'react';
import {Text} from 'react-native';
import {isAndroid} from '@helpers';

const DefaultFont = ({style, heading = false, children, ...props}) => {
  const textStyle = {
    fontFamily: isAndroid() ? 'sans-serif' : 'System',
    fontSize: heading ? 22 : 16,
  };
  let applyStyle;
  if (Array.isArray(style)) {
    applyStyle = [textStyle, ...style];
  } else {
    applyStyle = [textStyle, style];
  }

  return (
    <Text {...props} style={applyStyle}>
      {children}
    </Text>
  );
};

export {DefaultFont as Text};
