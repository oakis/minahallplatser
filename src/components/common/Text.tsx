import React from 'react';
import {StyleProp, Text, TextStyle} from 'react-native';
import {isAndroid} from '@helpers';

interface TextProps {
  style?: StyleProp<TextStyle>;
  heading?: boolean;
  children: React.ReactNode;
  onPress?: () => void;
}

const DefaultFont = ({
  style,
  heading = false,
  children,
  ...props
}: TextProps): JSX.Element => {
  const textStyle = {
    fontFamily: isAndroid() ? 'sans-serif' : 'System',
    fontSize: heading ? 22 : 16,
  };
  let applyStyle;
  if (Array.isArray(style)) {
    applyStyle = [textStyle, ...style];
  } else {
    applyStyle = [textStyle, style];
  }

  return (
    <Text {...props} style={applyStyle}>
      {children}
    </Text>
  );
};

export {DefaultFont as Text};
