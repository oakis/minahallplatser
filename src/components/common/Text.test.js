import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Text } from './Text';

it('should match snapshot without styles', () => {
    const wrapper = shallow(<Text />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('should match snapshot with array styles', () => {
    const wrapper = shallow(<Text style={[{ color: '#abc' }, { backgroundColor: '#def' }]} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
