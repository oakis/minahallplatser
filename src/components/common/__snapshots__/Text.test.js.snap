// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with array styles 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DefaultFont
    style={
      Array [
        Object {
          "color": "#abc",
        },
        Object {
          "backgroundColor": "#def",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "style": Array [
        Object {
          "fontFamily": "System",
          "fontSize": 16,
        },
        Object {
          "color": "#abc",
        },
        Object {
          "backgroundColor": "#def",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "style": Array [
          Object {
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#abc",
          },
          Object {
            "backgroundColor": "#def",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should match snapshot without styles 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DefaultFont />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
      "style": Array [
        Object {
          "fontFamily": "System",
          "fontSize": 16,
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
        "style": Array [
          Object {
            "fontFamily": "System",
            "fontSize": 16,
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
